src/v/pandaproxy/rest/handlers.cc

```diff
@@ -99,6 +98,9 @@ get_brokers(server::request_t rq, server::reply_t rp) {
 
           rp.mime_type = res_fmt;
           return std::move(rp);
+      })
+      .handle_exception_type([](const seastar::gate_closed_exception&) {
+          return server::reply_t{};
```

Comment:
The goal: the `invoke_on_cache` method enters a gate but does not handle the gate_closed_exception. So I'm handling that exception at the top-level calls here.

My thought process: We tend to ignore the gate_closed_exception and allow Redpanda to continue running which is representative of a 200 status. I think that is OK in the PP.
I can find another location to handle the gate_closed_exception if that is desired.

Reply:
What does this have to do with Redpanda?

This exception comes from the stopped client, yes? So if the client doesn't perform the request, why would a 200 status be the right thing to return? I don't see why the exception isn't just propagated, and all the existing error translation code used.

Reply:
> What does this have to do with Redpanda?

My apologies, that was bad writing on my part. This has nothing to do with Redpanda and more about how to handle the gate_closed_exception at shutdown. 

> This exception comes from the stopped client, yes? Or possibly also during shutdown? So if the client doesn't perform the request, why would a 200 status be the right thing to return? I don't see why the exception isn't just propagated to the exception_handler that does all the existing error translation code.

Yes this exception occurs at shutdown and gets reported to the user as 50003, so something like: `{error_code: 50003, msg: 'gate closed'}`. I guess the question here is, should users ignore that error? It seems the answer is yes.

Reply:
Spoke with Ben about this, we're dropping this commit. The user simply must retry the request if the error bubbles up to them.

Reply:
That exception propagates out of the handler because the `client` is being evicted and `stop` is called before the request is made.

It might be possible to add a semaphore or something to prevent eviction during the first client request so it can at least make some progress.

Reply:
> That exception propagates out of the handler because the `client` is being evicted and `stop` is called before the first request is completed.
> 
> It might be possible to add a semaphore or something to prevent eviction during the first client request so it can at least make some progress.

Done and this makes sense as to why the lock on the cache was helping so much!